axelor-sale/src/main/java/com/axelor/apps/sale/rest/dto/CurrencyResponse.java

```diff
@@ -0,0 +1,46 @@
+package com.axelor.apps.sale.rest.dto;
+
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Product;
+import com.axelor.utils.api.ObjectFinder;
+import com.axelor.utils.api.ResponseStructure;
+
+public class CurrencyResponse extends ResponseStructure {
+  private Long currencyId;
+  private String code;
+  private String name;
+
+  private String symbol;
+
+  public CurrencyResponse(Company company) {
+    super(ObjectFinder.NO_VERSION);
+    this.currencyId = company.getCurrency().getId();
+    this.code = company.getCurrency().getCode();
+    this.name = company.getCurrency().getName();
+    this.symbol = company.getCurrency().getSymbol();
+  }
+
+  public CurrencyResponse(Product product) {
+    super(ObjectFinder.NO_VERSION);
+    this.currencyId = product.getSaleCurrency().getId();
+    this.code = product.getSaleCurrency().getCode();
+    this.name = product.getSaleCurrency().getName();
+    this.symbol = product.getSaleCurrency().getSymbol();
+  }
```

Comment:
There should be only one constructor with Currency as parameter, we can create a method in a service to create the CurrencyResponse. This way we can null check currency in company and product.